.github/workflows/publish-release-image.yaml

```diff
@@ -0,0 +1,112 @@
+name: "[internal] Create and Publish Release Docker Image"
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      is_major_version:
+        required: true
+        type: boolean
+        default: false
+      timeout:
+        required: false
+        type: number
+        default: 35
+jobs:
+  create-docker-image:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        arch: [grayskull, wormhole_b0]
+    runs-on:
+      - build-docker
+      - in-service
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Download wheels
+        uses: actions/download-artifact@v4
+        with:
+          name: eager-dist-${{ matrix.os }}-${{ matrix.arch }}
+      - name: Get the name of the wheel and set up env variables
+        id: generate-tag-name
+        run: |
+          echo "WHEEL_FILENAME=$(ls -1 *.whl)" >> $GITHUB_ENV
+          REPO_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-metalium-${{ matrix.os }}-amd64-dev/${{ matrix.arch }}
+          echo "REPO_IMAGE_NAME=$REPO_IMAGE_NAME" >> $GITHUB_ENV
+          TAG_NAME=$REPO_IMAGE_NAME:${{ inputs.version }}
+          echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          build-args: |
+            WHEEL_FILENAME=${{ env.WHEEL_FILENAME }}
+            BASE_IMAGE_NAME=tt-metalium/${{ matrix.os }}-amd64
+          tags: ${{ env.TAG_NAME }}
+          context: .
+          file: dockerfile/release.Dockerfile
+  smoke-test-docker-image:
+    needs: create-docker-image
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        test_group:
+          [
+              {
+                arch: grayskull,
+                runs-on: ["cloud-virtual-machine", "E150", "in-service"],
+                cmd: pytest tests/end_to_end_tests,
+              },
+              {
+                arch: wormhole_b0,
+                runs-on: ["cloud-virtual-machine", "N150", "in-service"],
```

Comment:
Should we run on N300 too here?

Reply:
The test is super easy to add in this case. Given that these are smoke tests and we are testing the integrity of the container, does it make sense to run them on N300 as well? If so, super easy for me to add a line.

Reply:
Added!